a10_octavia/controller/worker/flows/a10_member_flows.py

```diff
@@ -48,61 +48,78 @@ def get_create_member_flow(self, topology):
+            provides=a10constants.VTHUNDER)
+        iface_flat_subflow = self.get_vthunder_interface_flat_subflow(a10constants.VTHUNDER,
+            enable_iface=True)
+        iface_vlan_subflow = self.get_vthunder_interface_vlan_subflow(a10constants.VTHUNDER)
+
+        create_member_flow.add(get_vthunder, iface_flat_subflow, iface_vlan_subflow)
+        create_member_flow.link(get_vthunder, iface_vlan_subflow,
+                                decider=self._create_new_subport_decider)
+        create_member_flow.link(get_vthunder, iface_flat_subflow,
+                                decider=self._create_new_nic_decider)
+
         # configure member flow for HA
         if topology == constants.TOPOLOGY_ACTIVE_STANDBY:
-            create_member_flow.add(a10_database_tasks.GetBackupVThunderByLoadBalancer(
-                name="get_backup_vThunder",
+            vthunder_network_subflow.add(a10_database_tasks.GetBackupVThunderByLoadBalancer(
+                name="get_{}".format(a10constants.BACKUP_VTHUNDER),
                 requires=constants.LOADBALANCER,
                 provides=a10constants.BACKUP_VTHUNDER))
-            create_member_flow.add(vthunder_tasks.AmphoraePostMemberNetworkPlug(
-                name="backup_amphora_network_plug",
-                rebind=[constants.ADDED_PORTS, constants.LOADBALANCER, a10constants.BACKUP_VTHUNDER]))
-            create_member_flow.add(vthunder_tasks.VThunderComputeConnectivityWait(
-                name="backup_compute_conn_wait",
-                rebind=[a10constants.BACKUP_VTHUNDER, constants.AMPHORA]))
-            create_member_flow.add(vthunder_tasks.EnableInterfaceForMembers(
-                name="backup_enable_interface",
-                rebind=[constants.ADDED_PORTS, constants.LOADBALANCER, a10constants.BACKUP_VTHUNDER]))
+            create_member_flow.add(self.get_vthunder_interface_subflow_flat(a10constants.BACKUP_VTHUNDER,
+                enable_iface=True))
 
         create_member_flow.add(handler_server.MemberCreate(
             requires=(constants.MEMBER, a10constants.VTHUNDER, constants.POOL)))
         create_member_flow.add(database_tasks.MarkMemberActiveInDB(
             requires=constants.MEMBER))
         create_member_flow.add(database_tasks.MarkPoolActiveInDB(
             requires=constants.POOL))
-        create_member_flow.add(database_tasks.
-                               MarkLBAndListenersActiveInDB(
-                                   requires=(constants.LOADBALANCER,
-                                             constants.LISTENERS)))
+        final_state = database_tasks.MarkLBAndListenersActiveInDB(
+                              name="member_mark_lb_and_listener_active",
+                              requires=(constants.LOADBALANCER,
+                                        constants.LISTENERS))
 
+        create_member_flow.add(final_state)
         return create_member_flow
 
+    def _create_new_nic_decider(self, history):
```

Comment:
You can club these `_create_new_nic_decider` and `_create_new_subport_decider` methods, right?

Reply:
Club them? We are actively using them.

Reply:
I meant, since both these methods have the same body definition, we can have a single decider method instead of two. But its alright, if these methods will serve different purposes afterwards.

Reply:
Ah yeah that's fair. I initially expected they'd have different logic. I'll remove one and rename the other.